### PR TITLE
re-enable tank suicide, but retain body and all objects

### PR DESCRIPTION
**Pull request by feemjmeem**
Controversial for some reason.

Lets people make a mess but retains all of their items and leaves an (ugly) corpse for debraining or chestbursting or headslugging or what have you.

### CODE REVIEW

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 1]
```diff
@@ -21,8 +21,25 @@
...
+			if (W)
+				W.loc = H.loc
+				W.dropped(H)
+				if(prob(50))
+					step(W, pick(alldirs))
+		H.hair_style = "Bald"
+		H.update_hair()
+		H.blood_max = 5
+		gibs(H.loc, H.viruses, H.dna)
+		H.adjustBruteLoss(9001) //to make the body super-bloody
```
Comment (PKPenguin321) [inline code/game/objects/items/weapons/tanks/tanks.dm]: :rotating_light: meme alert meme alert :rotating_light:

Comment (feemjmeem) [inline code/game/objects/items/weapons/tanks/tanks.dm]: It's not a new meme; it's a re-introduction of #11700, but fixing the issues referenced in #12159 and #12224

**Comment by feemjmeem** [comment]
As I describe in #12224, the joke is already there but is half-implemented due to the fix of the balance issues. This re-implements the functionality but fixes the issues. An alternative to this PR is removing it entirely as the joke makes no sense without a side-effect.

File: code/game/objects/items/weapons/tanks/tanks.dm (hunk 1 above)
Comment (duncathan) [inline code/game/objects/items/weapons/tanks/tanks.dm]: the meme is over 9000, not whatever your thing was

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 2]
```diff
@@ -21,8 +21,25 @@
 	var/volume = 70
 
 /obj/item/weapon/tank/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting the [src]'s valve to their lips! I don't think they're gonna stop!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
+	if (H && !qdeleted(H))
+		for(var/obj/item/W in (H.contents))
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: just ```in H```

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 3]
```diff
@@ -21,8 +21,25 @@
 	var/volume = 70
 
 /obj/item/weapon/tank/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting the [src]'s valve to their lips! I don't think they're gonna stop!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
+	if (H && !qdeleted(H))
+		for(var/obj/item/W in (H.contents))
+			H.unEquip(W)
+			if (H.client)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: cover in ```unEquip()```

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 4]
```diff
@@ -21,8 +21,25 @@
...
 
 /obj/item/weapon/tank/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting the [src]'s valve to their lips! I don't think they're gonna stop!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
+	if (H && !qdeleted(H))
+		for(var/obj/item/W in (H.contents))
+			H.unEquip(W)
+			if (H.client)
+				H.client.screen -= W
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: covered in ```unEquip()```

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 5]
```diff
@@ -21,8 +21,25 @@
...
+			if (W)
+				W.loc = H.loc
+				W.dropped(H)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: also covered in ```unEquip()```

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 6]
```diff
@@ -21,8 +21,25 @@
...
+	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting the [src]'s valve to their lips! I don't think they're gonna stop!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
+	if (H && !qdeleted(H))
+		for(var/obj/item/W in (H.contents))
+			H.unEquip(W)
+			if (H.client)
+				H.client.screen -= W
+			if (W)
+				W.loc = H.loc
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: covered in ```unEquip()```

File: code/game/objects/items/weapons/tanks/tanks.dm [hunk 7]
```diff
@@ -21,8 +21,21 @@
 	var/volume = 70
 
 /obj/item/weapon/tank/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='suicide'>[user] is putting the [src]'s valve to their lips! I don't think they're gonna stop!</span>")
 	playsound(loc, 'sound/effects/spray.ogg', 10, 1, -3)
+	if (H && !qdeleted(H))
+		spawn(0)
+		for(var/obj/item/W in H)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tanks/tanks.dm]: the for needs to be indented UNDER the ```spawn(0)```

**Comment by Firecage** [comment]
Honestly it would just be silly for it to spray gibs everywhere and leave behind a corpse. I would be in support to just remove the joke.

**Comment by CPTANT** [comment]
just gib the body and keep the items.

**Comment by Bawhoppen** [comment]
Thats what it did already, but people got mad. 
Also extracting blood from gibs when

**Comment by feemjmeem** [comment]
It doesn't spray body parts, just gibs. I used the generic gib spawner, not the human-specific one.

The extra damage to the body leaves the corpse a completely bloody mess in every limb. The way I see it, it's just a pile of guts in a loosely-bound skeleton shape. You're certainly not resuscitating it.

**Comment by Firecage** [comment]
The joke is bad to begin with though.

**Comment by feemjmeem** [comment]
:( You don't like Big Trouble in Little China?

**Pull request merged**